lua/scratch/config.lua

```diff
@@ -48,207 +56,80 @@ local default_config = {
         insert_mode = true,
       },
     },
-    ["gp.md"] = {
-      cursor = {
-        location = { 12, 2 },
-        insert_mode = true,
-      },
-      content = {
-        "# topic: ?",
-        "",
-        '- model: {"top_p":1,"temperature":0.7,"model":"gpt-3.5-turbo-16k"}',
-        "- file: placeholder",
-        "- role: You are a general AI assistant.",
-        "",
-        "Write your queries after 🗨:. Run :GpChatRespond to generate response.",
-        "",
-        "---",
-        "",
-        "🗨:",
-        "",
-      },
-    },
   },
-  ---@type Scratch.LocalKeyConfig[]
   localKeys = {
     {
-      filenameContains = { "gp" },
+      filenameContains = { "sh" },
       LocalKeys = {
         {
-          cmd = "<CMD>GpResponse<CR>",
-          key = "<C-k>k",
+          cmd = "<CMD>RunShellCurrentLine<CR>",
+          key = "<C-r>",
           modes = { "n", "i", "v" },
         },
       },
     },
   },
 }
 
-local function getConfigFilePath()
-  -- read CONFIG_FILE_PATH content as the filepath and return
-  local file = io.open(CONFIG_FILE_FLAG_PATH, "r")
-  local filepath = file:read("*all")
-  file:close()
+---@class Scratch.SetupConfig
+---@field json_config_path? string
+---@field scratch_config? Scratch.Config
 
-  return filepath
+local function editConfig()
+  vim.cmd(":e " .. vim.g.scratch_json_config_path)
 end
 
-local function validate_abspath(path)
-  -- Check if path contains any invalid characters
-  local valid_chars = "[^%w/%.%-%_]+"
-  if vim.fn.has("win32") == 1 then
-    valid_chars = "[^%w%.:%\\%-%_]+"
-  end
-  if string.match(path, valid_chars) then
-    return false
+---@return Scratch.Config
+local function get_and_update_config()
```

Comment:
Isn't `vim.g.scratch_config` already set during setup? Why have another function?

Reply:
If you want to change something during runtime, can't you just update it directly?

Reply:
I expose an API to allow user to manually update their json config during runtime~
So I hope the config will always have the up-to-date version~

Reply:
But this is not an API, this function is being used by every other function.

Having a function to read changes from JSON is fine, but then normal functions should just use `vim.g.scratch_config` directly.